client/src/lib.rs

```diff
@@ -1095,6 +1119,14 @@ impl WorkflowClientTrait for Client {
             binary_checksum: self.bound_worker_build_id.clone().unwrap_or_default(),
```

Comment:
You shouldn't send both binary checksum and build ID, decide which one to send based on server capabilities.